### PR TITLE
Create service repository controller

### PR DESCRIPTION
**Pull request by udodikdmitro**

### CODE REVIEW

File: src/main/java/com/catapi/controller/BreedController.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+public class BreedController {
+    private final BreedService breedService;
+
+    public BreedController(BreedService breedService) {
+        this.breedService = breedService;
+    }
+
+    @GetMapping("/update")
+    public ResponseEntity<String> updateBreedDbController() {
+        breedService.updateBreedDb();
```
Comment (HailToSG) [inline src/main/java/com/catapi/controller/BreedController.java]: I don't like the method name "updateBreedDb" and "updateBreedDbController"

It discribes nothing. It's much better to name it "updateBreedsFromExternalApi" in both cases (service and controller);

File: src/test/java/com/catapi/service/BreedServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    void requestBreeds_should_throw_exception_with_right_message() {
+        when(restTemplate.exchange(anyString(), any(), isNull(), any(ParameterizedTypeReference.class)))
+                .thenThrow(new RuntimeException("Помилка"));
+        ExternalApiException e = assertThrows(ExternalApiException.class, () -> breedService.getAllBreedsFromExternalApi());
+        assertEquals("Cannot get breed from external api: Помилка", e.getMessage());
+    }
+
+
+    @Test
+    void updateBreedDb_should_update_existing_breeds_and_add_new_ones() {
```
Comment (HailToSG) [inline src/test/java/com/catapi/service/BreedServiceTest.java:68]: This test doesn't prove that it claims in its name.

It only shows that "save" method was called 2 times.

Comment (HailToSG) [reply src/test/java/com/catapi/service/BreedServiceTest.java:68]: Use this:

ArgumentCaptor<Breed> breedCaptor = ArgumentCaptor.forClass(Breed.class);
Mockito.verify(breedRepository, Mockito.times(2)).save(breedCaptor.capture());
List<Breed> arguments = breedCaptor.getAllValues();

this is the way how you can get actual arguments from method which is being called.

Then you can get arguments from list (in this case is 

```
List<Breed> arguments
```

) and compare what you've got with what you expect

Comment (HailToSG) [reply src/test/java/com/catapi/service/BreedServiceTest.java:68]: how to compare? It was supposed that you will do something like that:

```
Breed expectedUpdated = new Breed();
expectedUpdated.setOuterBreedId(existingBreedEntity.getOuterBreedId());
expectedUpdated.setBreedName(breedView1.name());
expectedUpdated.setDescription(breedView1.description());
assertEquals(expectedUpdated, savedBreeds.get(0));
```

It shows that your first saved breed was updated indeed (some fields recieved values from BreedView1).

File: src/main/java/com/catapi/controller/BreedController.java (hunk 1 above)
Comment (udodikdmitro) [reply src/main/java/com/catapi/controller/BreedController.java]: Done

File: src/test/java/com/catapi/service/BreedServiceTest.java (hunk 2 above)
Comment (udodikdmitro) [reply src/test/java/com/catapi/service/BreedServiceTest.java:68]: Done

**Review by HailToSG — APPROVED** [review]

**Pull request merged**